src/Microsoft.OData.Core/UriParser/ExpressionLexer.cs

```diff
@@ -1195,7 +1195,7 @@ private void AdvanceThroughBalancedExpression(char startingCharacter, char endin
         /// <summary>Parses an identifier by advancing the current character.</summary>
         private void ParseIdentifier()
         {
-            Debug.Assert(this.IsValidStartingCharForIdentifier, "Expected valid starting char for identifier");
+            Debug.Assert(this.IsValidStartingCharForIdentifier || this.ch == '@', "Expected valid starting char for identifier");

```

Comment:
Maybe provide a function to check for '@' instead of hard-coding '@' in a bunch of places? Here and ExpressionLexer.cs.. I say function because the intent of the check would be expressed in the function name, making the conditional more readable.

Reply:
Note that, in some cases we are comparing to '@' to determine if it's an annotation; in other cases we're trying to determine if it's a parameter alias.  I have an "IsAnnotation" method which also verifies that it contains a ".", but that can't be used here because we only have the single char.  I have updated all cases where we are comparing a character to "@" to use the new AnnotationPrefix constant, which provides context as to what we are checking for.

---
In reply to: [167023686](https://github.com/OData/odata.net/pull/1063#discussion_r167023686) [](ancestors = 167023686)